chain/client/src/stateless_validation/chunk_endorsement_tracker.rs

```diff
@@ -186,16 +167,25 @@ impl ChunkEndorsementTracker {
             chunk_header.shard_id(),
             chunk_header.height_created(),
         )?;
-        // Get the chunk_endorsements for the chunk from our cache.
-        // Note that these chunk endorsements are already validated as part of process_chunk_endorsement.
+
+        // Get the chunk_endorsements for the chunk from chunk_endorsements and pending_chunk_endorsements cache.
+        // Note that these chunk endorsements in chunk_endorsements are already validated.
         // We can safely rely on the following details
         //    1. The chunk endorsements are from valid chunk_validator for this chunk.
         //    2. The chunk endorsements signatures are valid.
-        let Some(chunk_endorsements) = self.chunk_endorsements.get(&chunk_header.chunk_hash())
-        else {
-            // Early return if no chunk_endorsements found in our cache.
-            return Ok(ChunkEndorsementsState::NotEnoughStake(None));
-        };
+        // For pending_chunk_endorsements, we would validate the chunk endorsements before including them in the block.
+        let mut chunk_endorsements =
+            self.chunk_endorsements.get(&chunk_header.chunk_hash()).unwrap_or_default();
+
+        for (account_id, endorsement) in
+            self.pending_chunk_endorsements.get(&chunk_header.chunk_hash()).unwrap_or_default()
```

Comment:
We should also cleanup pending endorsements to avoid processing them again and again.